lib/inv_bicgstabl_quda.cpp

```diff
@@ -533,86 +720,35 @@ namespace quda {
         
         // r[j+1] = A r[j], x = x + alpha*u[0]
         matSloppy(*r[j+1], *r[j], temp);
-	dslash::aux_worker = NULL;
+        dslash::aux_worker = NULL;
 
       } // End BiCG part.
 
-      // MR part. Really just modified Gram-Schmidt.
-      // The algorithm uses the byproducts of the Gram-Schmidt to update x
-      //   and other such niceties. One day I'll read the paper more closely.
-      // Can take this from 'orthoDir' in inv_gcr_quda.cpp, hard code pipelining up to l = 8.
-      for (int j = 1; j <= n_krylov; j++) {
+      // Perform the MR portion of BiCGstab-L
+      // if we're doing a fixed number of iterations, we only need to update x
+      computeMR(x_sloppy, fixed_iteration);
```

Comment:
It should be made more obvious, e.g., wrapped in an `#ifdef` and commented, how to disable the new communication-avoiding path and enable the old path here.

Reply:
https://github.com/lattice/quda/commit/4ca7a56414aba75a68ece64632efa50abd4a41e4